pkg/api/ref.go

```diff
@@ -58,16 +59,18 @@ func GetReference(obj runtime.Object) (*ObjectReference, error) {
 	}
 	version := ""
 	parsedSelfLink := versionFromSelfLink.FindStringSubmatch(meta.SelfLink())
-	if len(parsedSelfLink) < 2 {
-		if ForTesting_ReferencesAllowBlankSelfLinks {
-			version = "testing"
-		} else if meta.SelfLink() == "" {
-			return nil, ErrNoSelfLink
+	if mustValidateSelfLink(obj) {
```

Comment:
@smarterclayton I am pretty convinced that all of the changes to ref.go can go away in this PR.  I made a mistake in #4974 that caused the integration tests to fail with a panic in this PR.  I'm going to fix it, and then we need to work through the self-link issue.